streaming/base/dataset.py

```diff
@@ -163,7 +163,7 @@ def __init__(self,
         self._partition_state = None
 
         # Seed is set below.
-        world = World()
+        self._rank_world = world = World()
```

Comment:
add a comment explaining the footgun here

Reply:
Agreed, not clear to me what the fix is here.  Is there a race condition that the chained assignment is solving?